ext/standard/array.c

```diff
@@ -5935,7 +5935,8 @@ PHP_FUNCTION(array_sum)
 				ZVAL_LONG(&tmp, 0);
 				add_function(return_value, return_value, &tmp);
 			}
-			// TODO Warning/Deprecation?
+			php_error_docref(NULL, E_WARNING, "Addition is not supported on type %s",
```

Comment:
I'd vote for a deprecation + removal in PHP 9.0 as the current behavior is basically an oversight since arithmetic operations are already not supported on resources and other types, and your changes are pretty much in line with Nikita's RFC.